### ISSUE TITLE
Multivariate normal distribution crashes on broadcast

### ISSUE THREAD

**Issue by jlowin** [tensorflow/tensorflow#3054]
### Environment info
Operating System: Mac OS X 10.11.5

If installed from binary pip package, provide:

1. Which pip package you installed. 
  https://storage.googleapis.com/tensorflow/mac/tensorflow-0.9.0-py3-none-any.whl

2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.
  0.9.0

### Issue

The Multivariate Normal distribution in `contrib.distributions` [claims to support broadcasting](https://github.com/tensorflow/tensorflow/blob/d42facc3cc9611f0c9722c81551a7404a0bd3f6b/tensorflow/contrib/distributions/python/ops/mvn.py#L378) but it does not appear to work. This script will illustrate the issue:

```python
import numpy as np
import tensorflow as tf

# dimensions
a = 2
b = 3
c = 4

mu = tf.reshape(tf.constant(
    np.arange(a * b * c * 1.0)),
    [a, b, c])

sigma = tf.batch_matrix_diag(tf.reshape(tf.constant(
    np.arange(a * b * c * 1.0) + 1),
    [a, b, c]))

mvn = tf.contrib.distributions.MultivariateNormal(mu, sigma)

with tf.Session() as sess:
    # works because shape(input) == shape(mu)
    sess.run(mvn.pdf(np.ones((a, b, c))))
    
    # fails even though input and mu are broadcastable
    sess.run(mvn.pdf(np.ones((1, a, b, c))))
```

**Comment by langmore** [comment]
Hello Andy,

Thanks for bringing this to our attention.  I verified that this operation indeed raises an error during the broadcast attempt.  Just today we have updated the MultivariateNormal API ([PR](https://github.com/tensorflow/tensorflow/commit/94cbf42a074128f9c46d980f9fddb47fba3602a6)).  I verified that your example works in the new API.  For your reference, in the new API you would do:

```python
mvn = tf.contrib.distributions.MultivariateNormalFull(mu, sigma)
```

This defines an MVN via the full covariance matrix.  Also see `MultivariateNormalCholesky`.  I'll add `MultivariateNormalDiag` soon as well.
